src/server/handler/authMiddleware.ts

```diff
@@ -0,0 +1,40 @@
+import * as express from "express";
+
+import { Config } from "../../config";
+import * as utils from "../utils";
+import Session from "../types/session";
+
+export default class AuthMiddleware {
+
+    public static auth(permissionName: string): express.Handler {
+        const namespacedPermission = Config.instance.permissionNamespace + "." + permissionName;
+        return async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
+            // get cookie from frontend
+            const sid = req.cookies.HOPPER_SESSION;
+            if (sid == undefined) {
+                console.log("Cookie undefined");
```

Comment:
do we really want to print this any time someone tries to to open hopper without a session?